stackit/internal/services/cdn/customdomain/resource.go

```diff
@@ -202,7 +251,7 @@ func (r *customDomainResource) Read(ctx context.Context, req resource.ReadReques
 		core.LogAndAddError(ctx, &resp.Diagnostics, "Error reading CDN custom domain", fmt.Sprintf("Calling API: %v", err))
 		return
 	}
-	err = mapCustomDomainFields(customDomainResp.CustomDomain, &model)
+	err = mapCustomDomainFields(customDomainResp.CustomDomain, &model, customDomainResp.Certificate)
```

Comment:
based on the previous comments, this change will be required
```suggestion
	err = mapCustomDomainFields(customDomainResp, &model)
```